standalone/pruntime/pruntime/src/main.rs

```diff
@@ -0,0 +1,159 @@
+struct Args {
+    /// Number of CPU cores to be used for mining.
+    #[structopt(short, long)]
+    cores: Option<u32>,
+
+    /// Run benchmark at startup.
+    #[structopt(long)]
+    init_bench: bool,
+
+    #[structopt(long, default_value = "./GeoLite2-City.mmdb")]
+    geoip_city_db: String,
+
+    /// Allow CORS for HTTP
+    #[structopt(long)]
+    allow_cors: bool,
+
+    /// Turn on /kick API
+    #[structopt(long)]
+    enable_kick_api: bool,
+
+    /// Log filter passed to env_logger
+    #[structopt(long, default_value = "INFO")]
+    log_filter: String,
+
+    /// Listening IP address of HTTP
+    #[structopt(long)]
+    address: Option<String>,
+
+    /// Listening port of HTTP
+    #[structopt(long)]
+    port: Option<String>,
+}
+
+fn main() {
+    // Disable the thread local arena(memory pool) for glibc.
+    // See https://github.com/gramineproject/gramine/issues/342#issuecomment-1014475710
+    #[cfg(target_env = "gnu")]
+    unsafe {
+        libc::mallopt(libc::M_ARENA_MAX, 1);
+    }
+
+    let runing_under_gramine = std::path::Path::new("/dev/attestation/user_report_data").exists();
+    let sealing_path = if runing_under_gramine {
+        // In gramine, the protected files are configured via manifest file. So we must not allow it to
+        // be changed at runtime for security reason. Thus hardcoded it to `/protected_files` here.
+        // Should keep it the same with the manifest config.
+        "/protected_files"
+    } else {
+        "."
+    }
+    .into();
+
+    let args = Args::from_args();
+
+    env::set_var("RUST_BACKTRACE", "1");
+    env::set_var("ROCKET_ENV", "dev");
+
+    if let Some(address) = &args.address {
+        env::set_var("ROCKET_ADDRESS", address);
+    }
+
+    if let Some(port) = &args.port {
+        env::set_var("ROCKET_PORT", port);
+    }
+
+    let env = env_logger::Env::default().default_filter_or(&args.log_filter);
+    env_logger::Builder::from_env(env).init();
+
+    let init_args = InitArgs {
+        sealing_path,
+        log_filter: Default::default(),
+        init_bench: args.init_bench,
+        version: env!("CARGO_PKG_VERSION").into(),
+        git_revision: git_revision(),
+        geoip_city_db: args.geoip_city_db,
+        enable_checkpoint: false,
+        checkpoint_interval: 0,
+        skip_corrupted_checkpoint: false,
+    };
+    info!("init_args: {:#?}", init_args);
+    if let Err(err) = runtime::ecall_init(init_args) {
+        panic!("Initialize Failed: {:?}", err);
+    }
+
+    let bench_cores: u32 = args.cores.unwrap_or_else(|| num_cpus::get() as _);
+    info!("Bench cores: {}", bench_cores);
+
+    let rocket = thread::Builder::new()
+        .name("rocket".into())
+        .spawn(move || {
+            let err = api_server::rocket(args.allow_cors, args.enable_kick_api).launch();
+            panic!("Launch rocket failed: {}", err);
+        })
+        .expect("Failed to launch Rocket");
+
+    let mut v = vec![];
+    for i in 0..bench_cores {
```

Comment:
Need to check if the performance score matches it in SGX SDK

Reply:
OK, let me have a check.

Reply:
Checked on my machine. There is no significant difference. Sometimes gramine is slightly better.
![image](https://user-images.githubusercontent.com/6442159/153360921-cc7b08ee-1704-4d17-b648-9afb31c58579.png)